src/model/model.py

```diff
@@ -98,31 +98,25 @@ def save(self, data, path):
             imageio.imwrite(path, data)
         return "saved as {}".format(path)
 
-    def extract_channel(self, im, red=True, green=True, blue=True, alpha=True):
+    def extract_channel(self, im, channel='red'):
         """
         extract channel from image
 
         Parameters
         ----------
         im: 2D numpy array
-        red, green, blue, alpha: bool, default=True
-            if False, drop this channel
+        channel: {'red', 'green', 'blue'}, default='red'
 
         Return
         ------
         result: 2D numpy array
 
         """
-        if im.shape[2] == 4:
-            selected_channels = [i for i, ch in enumerate([red, green, blue, alpha]) if ch]
-        elif im.shape[2] == 3:
-            selected_channels = [i for i, ch in enumerate([red, green, blue]) if ch]
-        if len(selected_channels) == 1:
-            selected_channels = selected_channels[0]
-        elif len(selected_channels) == 0:
-            raise ValueError("No channel has been selected")
-        im = im[:, :, selected_channels]
-        return im
+        if im.ndim != 3 or im.shape[2] != 3:
+            raise Exception("You cannot extract red, green or blue channel from this image")
+        for i, c in enumerate(['red', 'green', 'blue']):
+            if c == channel:
+                return im[:, :, i]
```

Comment:
Just a proposition to make it more compact :

    idx = ['red', 'green', 'blue'].index(channel)
    return I'm[:, :, idx]

Reply:
ure damn right